api/features/models.py

```diff
@@ -220,12 +227,17 @@ class FeatureSegment(
     # specific attributes for managing the order of feature segments
     priority = models.PositiveIntegerField(editable=False, db_index=True)
     order_field_name = "priority"
-    order_with_respect_to = ("feature", "environment")
+    order_with_respect_to = ("feature", "environment", "environment_feature_version")
 
     objects = FeatureSegmentManager()
 
     class Meta:
-        unique_together = ("feature", "environment", "segment")
+        unique_together = (
+            "feature",
+            "environment",
+            "segment",
+            "environment_feature_version",
```

Comment:
The `sqlmigrate` for the migration for this (and the new field) is as follows: 

```
BEGIN;
--
-- Add field environment_feature_version to featurestate
--
ALTER TABLE "features_featurestate" ADD COLUMN "environment_feature_version_id" uuid NULL CONSTRAINT "features_featurestat_environment_feature__0b08c62e_fk_feature_v" REFERENCES "feature_versioning_environmentfeatureversion"("uuid") DEFERRABLE INITIALLY DEFERRED; SET CONSTRAINTS "features_featurestat_environment_feature__0b08c62e_fk_feature_v" IMMEDIATE;
--
-- Add field environment_feature_version to historicalfeaturestate
--
ALTER TABLE "features_historicalfeaturestate" ADD COLUMN "environment_feature_version_id" uuid NULL;
CREATE INDEX "features_featurestate_environment_feature_version_id_0b08c62e" ON "features_featurestate" ("environment_feature_version_id");
CREATE INDEX "features_historicalfeature_environment_feature_versio_5d9a2014" ON "features_historicalfeaturestate" ("environment_feature_version_id");
COMMIT;
```

I think the modification of the index will cause the table to be locked for a very short amount of time, however, since we only have 17k records in the FeatureSegment table, I think this is an acceptable risk. Keen to know what you think here though @gagantrivedi @khvn26 ?

Reply:
I don't think any of them should lock the table? I think create index by default is not concurrently?

Reply:
Sorry, that's the wrong sql migrate 🤦 this is the right one: 

```sql
BEGIN;
--
-- Add field environment_feature_version to featuresegment
--
ALTER TABLE "features_featuresegment" ADD COLUMN "environment_feature_version_id" uuid NULL CONSTRAINT "features_featuresegm_environment_feature__11efe443_fk_feature_v" REFERENCES "feature_versioning_environmentfeatureversion"("uuid") DEFERRABLE INITIALLY DEFERRED; SET CONSTRAINTS "features_featuresegm_environment_feature__11efe443_fk_feature_v" IMMEDIATE;
--
-- Add field environment_feature_version to historicalfeaturesegment
--
ALTER TABLE "features_historicalfeaturesegment" ADD COLUMN "environment_feature_version_id" uuid NULL;
--
-- Alter unique_together for featuresegment (1 constraint(s))
--
ALTER TABLE "features_featuresegment" DROP CONSTRAINT "features_featuresegment_feature_id_environment_i_e5499ca3_uniq";
ALTER TABLE "features_featuresegment" ADD CONSTRAINT "features_featuresegment_feature_id_environment_i_f1fde686_uniq" UNIQUE ("feature_id", "environment_id", "segment_id", "environment_feature_version_id");
CREATE INDEX "features_featuresegment_environment_feature_version_id_11efe443" ON "features_featuresegment" ("environment_feature_version_id");
CREATE INDEX "features_historicalfeature_environment_feature_versio_88d80be3" ON "features_historicalfeaturesegment" ("environment_feature_version_id");
COMMIT;
```

Specifically, it's the `DROP CONSTRAINT` / `ADD CONSTRAINT` that I'm worried about.